### PR TITLE
Populate existing collections by default when deserializing JSON

### LINKED ISSUE #16585: Breaking shape name resolution regarding menu shape
### Describe the bug

I found this undocumented breaking change regarding the "menu" shape (potentially all non PascalCase shapes).

### Orchard Core version

Latest `main`.

### To Reproduce

1. Set up an OC 2.0 site with any recipe that creates a `main-menu` content item (e.g. Blog).
2. Create a content type with a Liquid part.
3. Create a content item with the Liquid content `{% shape "menu", alias: "alias:main-menu" %}`
4. Go to Admin > Design > Templates.
5. Create a template called "Menu" with any content.
6. Display the content item, observe that the shape is not overridden.
7. Delete the "Menu" template and create a "menu" template instead.
8. Display the content item again, observe that the shape is successfully overridden.
9. Now do the same again in OC 1.8, observe that in 1.8 the menu shape is overridden by the "menu" template but not the "Menu" template, in other words it's the opposite.

| override works | menu | Menu |
|--------|------|------|
| OC 1.8 | ❌    | ✅    |
| OC 2.0 | ✅    | ❌    |

Additional note: btw you can't create `menu` and `Menu` templates at the same time in Admin > Design > Templates. You have to delete one first to create the other. Is this intentional? Seems like the template creation is checked case-insensitively, but the template evaluation is case-sensitive.

### Expected behavior

Either "Menu" template override should keep working. It would be even better if both "Menu" and "menu" worked. Maybe it should first try the shape type case sensitively and if it's not found then case insensitively? (that's what I do with media library themes in Lombiq.Hosting.MediaTheme [here](https://github.com/Lombiq/Hosting-Media-Theme/blob/7fb210ebb07516a61fbf4d4f3b110b33436ed6a6/Lombiq.Hosting.MediaTheme.Bridge/Services/MediaThemeManager.cs#L84-L95))

If that's not possible, then please document the breaking change in the [2.0.0.md](https://github.com/OrchardCMS/OrchardCore/blob/3fe9612adb08b03114177a485ecc02e52091030c/src/docs/releases/2.0.0.md?plain=1) release notes.

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #16585

@sarahelsaig

### CODE REVIEW

**Comment by sebastienros** [comment]
I found that the sessions didn't have this property. Isolated it to these defaults. I want to see if the tests still pass as we might have not set it on purpose, I remember we had issues that STJ was not able to assign new collections and we went with `init` setters for this reason. But I believe that if it find existing collections it will just be fine.

**Comment by MikeAlhayek** [comment]
Strange that we did not add it. I think it should be added too. I suggest that you add a test case to approve that this fixed the problem we had.

**Review by MikeAlhayek — APPROVED** [review]

**Comment by gvkries** [comment]
This is way better than adding another `JsonConverter`. 
Should it also be set in `JsonSerializerOptionsExtensions.Merge()`?

https://github.com/OrchardCMS/OrchardCore/blob/e23ab9e101b392cc6fd313c9da94fd74de7d5bb1/src/OrchardCore/OrchardCore.Abstractions/Json/Extensions/JsonSerializerOptionsExtensions.cs#L10

File: src/OrchardCore/OrchardCore.Abstractions/Extensions/DocumentJsonSerializerOptionsConfiguration.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public void Configure(DocumentJsonSerializerOptions options)
         options.SerializerOptions.PropertyNameCaseInsensitive = JOptions.Base.PropertyNameCaseInsensitive;
         options.SerializerOptions.AllowTrailingCommas = JOptions.Base.AllowTrailingCommas;
         options.SerializerOptions.WriteIndented = JOptions.Base.WriteIndented;
+        options.SerializerOptions.PreferredObjectCreationHandling = JsonObjectCreationHandling.Populate;
```
Comment (sarahelsaig) [inline src/OrchardCore/OrchardCore.Abstractions/Extensions/DocumentJsonSerializerOptionsConfiguration.cs:25]: Why not this?
```suggestion
        options.SerializerOptions.PreferredObjectCreationHandling = JOptions.Base.PreferredObjectCreationHandling;
```
As I mentioned in the other PR, this setting is already applied to Base.

**Review by sarahelsaig — CHANGES_REQUESTED** [review]
I agree with @gvkries , `PreferredObjectCreationHandling` should be added to `JsonSerializerOptionsExtensions.Merge()` and that should be used here instead of manually overwriting each property.
That way you will never have to manually update this configuration from `JOptions.Base` if a new setting is added there or something.

File: src/OrchardCore/OrchardCore.Abstractions/Extensions/DocumentJsonSerializerOptionsConfiguration.cs (hunk 1 above)
Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Abstractions/Extensions/DocumentJsonSerializerOptionsConfiguration.cs]: Didn't realize the other lines were copying the base options.

**Pull request closed without merging**